test/testsuite/linalg.jl

```diff
@@ -15,9 +15,20 @@
         @test compare(x -> permutedims(x, (2, 1, 3)), AT, rand(Float32, 4, 5, 6))
         @test compare(x -> permutedims(x, (3, 1, 2)), AT, rand(Float32, 4, 5, 6))
         @test compare(x -> permutedims(x, [2,1,4,3]), AT, randn(ComplexF32,3,4,5,1))
+        AT <: GPUArrays.AbstractGPUArray && @test let
+            x = randn(ComplexF32,3,4,5,1)
+            y = permutedims(x, (2,1,4,3))
+            Array(GPUArrays._permutedims!(UInt64, AT(zero(y)), AT(x), (2,1,4,3))) ≈ y
```

Comment:
Why does this test an non-public function?

Reply:
I need to test UInt64 version to make sure it works properly when array length is larger than typemax of UInt32. It is very like to blow the gpu memory of the test ci.

Reply:
OK, fair enough. Could you add that as a comment?

Reply:
Done